app/models/queue_item.rb

```diff
@@ -5,12 +5,28 @@ class QueueItem < ActiveRecord::Base
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
 
+  validates_numericality_of :position, :only_integer =>  true
+
   def rating
-    review = Review.where(user_id: user.id, video_id: video.id).first
     review.rating if review
   end
 
+  def rating=(new_rating)
+    if review
+      review.update_column(:rating, new_rating)
+    else
+      review = Review.new(user: user, video: video, rating: new_rating)
+      review.save(validate: false)
+    end
+  end
+
   def category_name
     category.name
   end
+
+
+  def review
+    @review ||= Review.find_by(user_id: user.id, video_id: video.id)
+  end
+  private
```

Comment:
if `private` keyword is at the bottom, then no method is private. Only those that are below it in code are private.